adapters/repos/db/migrator.go

```diff
@@ -253,6 +253,63 @@ func (m *Migrator) RecalculateVectorDimensions(ctx context.Context) error {
 	return nil
 }
 
+func (m *Migrator) RecountProperties(ctx context.Context) error {
+	count := 0
+	m.logger.
+		WithField("action", "recount").
+		Info("Recounting properties, this may take a while")
+
+	clearedShard := map[string]bool{}
+
+	// Iterate over all indexes
+	for _, index := range m.db.indices {
+		// Iterate over all shards
+		err := index.IterateObjects(ctx, func(index *Index, shard *Shard, object *storobj.Object) error {
+			if _, ok := clearedShard[shard.name]; !ok {
+				shard.propLengths.Clear()
```

Comment:
I'd use `index.IterateShards` to clear them all out before iterating over all objects